### ISSUE TITLE
Consider performance impact of attribute-based default relays

### ISSUE THREAD

**Issue by adamchester** [AutoFixture/AutoFixture#221]
During my performance testing (e.g. #218), I discovered that the `ISpecimenBuilder` relays, which query for attributes via an `ICustomAttributeProvider`, have a major impact on performance. The negative impact exists regardless if the caller/user requires them.

My testing revealed that by commenting out [these lines from `DefaultRelays`](https://github.com/AutoFixture/AutoFixture/blob/04c0d4c0/Src/AutoFixture/DefaultRelays.cs#L25-L28) (i.e. removing the `StringLengthAttributeRelay`, `RangeAttributeRelay`, and `RegularExpressionAttributeRelay`) resulted in a huge performance improvement.

For example, with some simple tests I used in #218, the following was observed:

```
WITH-ATTR-RELAYS>msbuild BuildRelease.msbuild /t:test /m:8 | find /i /n "quick"
[3787]             Ploeh.AutoFixtureUnitTest.FixtureTest.Create10000ImmutableSpecimensQuickly
[3789]               Create10000ImmutableSpecimensQuickly, Elapsed: 00:00:00.6987564
[3802]             Ploeh.AutoFixtureUnitTest.FixtureTest.Create1000SpecimensQuickly
[3804]               Create1000SpecimensQuickly, Elapsed: 00:00:01.5871797
[3840]             Ploeh.AutoFixtureUnitTest.FixtureTest.InstantiateFixtureQuickly
[3842]               InstantiateFixtureQuickly, Elapsed: 00:00:00.0291962

NO-ATTR-RELAYS>msbuild BuildRelease.msbuild /t:test /m:8 | find /i /n "quick"
[3553]             Ploeh.AutoFixtureUnitTest.FixtureTest.Create10000ImmutableSpecimensQuickly
[3555]               Create10000ImmutableSpecimensQuickly, Elapsed: 00:00:00.3815511
[3713]             Ploeh.AutoFixtureUnitTest.FixtureTest.Create1000SpecimensQuickly
[3715]               Create1000SpecimensQuickly, Elapsed: 00:00:00.6354635
[3780]             Ploeh.AutoFixtureUnitTest.FixtureTest.InstantiateFixtureQuickly
[3782]               InstantiateFixtureQuickly, Elapsed: 00:00:00.0260725
```

In short, for `Create1000SpecimensQuickly`, WITH-ATTR-RELAYS = 1.58 seconds, NO-ATTR-RELAYS = 0.63 seconds. That's **more than 50% difference**.

The most obvious solution is to make it easy to turn them off. Does anyone have a better idea?
